mailbox/store/src/main/java/org/apache/james/mailbox/store/StoreRightManager.java

```diff
@@ -96,7 +96,11 @@ public boolean hasRight(Mailbox mailbox, Right right, MailboxSession session) {
     @Override
     public Rfc4314Rights myRights(MailboxPath mailboxPath, MailboxSession session) throws MailboxException {
         if (mailboxPath.belongsTo(session)) {
-            return MailboxACL.FULL_RIGHTS;
+            if (mailboxSessionMapperFactory.getMailboxMapper(session).pathExists(mailboxPath).block()) {
```

Comment:
are you adding one request to remove one?

Reply:
No. Insteadof 2 request (mbx+acl) we only do one (mbx). This is a net win.